src/navigator/navigator.js

```diff
@@ -49,62 +49,66 @@ var navigatorFactory = function navigatorFactory(testContext, testMap) {
+                        .catch(() => reject);
+                }
+                
+                reject();
+            });
 
-            if (_.isFunction(this[methodName])) {
-                return this[methodName](position);
-            }
         },
 
         /**
          * Navigate to the next item
-         * @returns {Object} the new test context
+         * @returns {Promise} the new test context
          */
         nextItem: function nextItem() {
-            return testContextBuilder.buildTestContextFromPosition(
-                testContext,
-                testMap,
-                testContext.itemPosition + 1
-            );
+            return this.jumpItem(testContext.itemPosition + 1);
         },
 
         /**
          * Navigate to the next item
-         * @returns {Object} the new test context
+         * @returns {Promise} the new test context
          */
         previousItem: function previsousItem() {
-            return testContextBuilder.buildTestContextFromPosition(
-                testContext,
-                testMap,
-                testContext.itemPosition - 1
-            );
+            return this.jumpItem(testContext.itemPosition - 1);
         },
 
         /**
          * Navigate to the next item
-         * @returns {Object} the new test context
+         * @returns {Promise} the new test context
          */
         nextSection: function nextSection() {
             var sectionStats = mapHelper.getSectionStats(testMap, testContext.sectionId);
             var section = mapHelper.getSection(testMap, testContext.sectionId);
 
-            return testContextBuilder.buildTestContextFromPosition(
-                testContext,
-                testMap,
-                section.position + sectionStats.total
-            );
+            return this.jumpItem(section.position + sectionStats.total);
         },
 
         /**
          * Navigate to the given position
          * @param {Number} position - the position
-         * @returns {Object} the new test context
+         * @returns {Promise} the new test context
          */
         jumpItem: function jumpItem(position) {
-            return testContextBuilder.buildTestContextFromPosition(testContext, testMap, position);
+            var item = mapHelper.getItemAt(updatedMap, position);
+            if (item && itemStore.has(item.id)) {
+                return itemStore.get(item.id)
+                    .then(itemFromStore => {
+                        const newTestContext = testContextBuilder.buildTestContextFromPosition(testContext, testMap, position, itemFromStore.attempt)
+                        return itemStore.update(newTestContext.itemIdentifier, 'attempt', newTestContext.attempt)
+                            .then(() => newTestContext);
+                    })
+            }
+            return Promise.resolve(testContextBuilder.buildTestContextFromPosition(testContext, testMap, position));
```

Comment:
So the only reason why you need to return a promise is because of the itemStore?
In that case, you don't need it. There is no itemStore in the normal mode.